### DISCUSSION TITLE
About ineritance

### DISCUSSION THREAD

**Discussion by hput** [psy0rz/zfs_autobackup#135] [Q&A]
I've had a problem several times now of pulling replication backup.  (These are practice runs .. no important data is being exposed to it.)

I know there is such a thing as inheritance but what to expect in practice seems unpredictable.
If you don't have full knowledge of how it works.

It seemed at one point while rescuing a VM I had put in jeopardy.   To rescue the vm I tried to use zfs set mountpoint=none  I found that it apparently cannot be expected to be inherited down a chain of zfs fs lke pool/seg1/seg2 ... etc;  I zfs set mountpoint=none at pool/seg1/  On an already implaced backup  already on the backup host.  I see the "zfs set" action take effect at the specific spot but does not migrate on down the chain.  Maybe I'm expecting something that is asking too much. 

In another effort I used canmount=noauto put in place before the replication backup was run.
That is,  the receiver fs put in play with a rendition of the example in "get started"
zfs-autobackup -v --ssh-source pve01 offsite1 data/backup/pve01 ... 

Before sending it off to start pulling  I did "zfs set canmount=noauto data/backup/pve01
Expecting that any backups stuck on that chain would inherit the canmount=noauto setting.
They didn't and instead retained their settings for mountpoint and canmount setting is "on" at
 data/backup/pve01/bkup1/ and at any more segments that go there during the pull operation.

So  mountpoints can still be threat to indigenous mountpoints on the backup server.
After cleaning up the vm I plan to try setting mountpoint=none at data/backup/pve01
And finally I will try adding  --clear-mountpoint like:

zfs-autobackup -v --clear-mountpoint --ssh-source pve01 offsite1 data/backup/pve01

Trying to see the best way to do this thru experimentation.

**Comment by Scrin** [comment]
The inheritance behavior depends on the property, some properties inherit with the original value (such as `compression`), some properties inherit a value that's different, but based on the parent value (such as `mountpoint`), and some properties are not inheritable at all (such as `canmount`). This is ZFS specific behavior (and not related zfs_autobackup) and there could be subtle differences between different ZFS implementations, but for example the [oracle docs regarding ZFS properties](https://docs.oracle.com/cd/E36784_01/html/E36835/gazss.html) state for the canmount property that "This property can be set on any file system, and the property itself is not inheritable."
